### ISSUE TITLE
Travis CI: rustup-init.sh for a nightly no longer works without clippy

### ISSUE THREAD

**Issue by RalfJung** [rust-lang/rustup#2063]
**Problem**

My Travis CI config contains the following command
```
curl https://build.travis-ci.org/files/rustup-init.sh -sSf | sh -s -- -y --default-toolchain nightly
```
Travis is still on rustup 1.19 (at least when I download it locally; not sure which version CI uses), so I cannot use the new `--profile` flag on `rustup-init.sh`.

**Steps**

1. Run the above command.
2. It fails, because clippy is missing.

**Further information**

Until a few days ago, that command was guaranteed to always work. But it looks like something now changed that makes even old rustup's use the new `default` profile. Maybe that change was done too quickly?

However, even once Travis updates their rustup, the command above can fail when clippy is missing from the current nightly. That's a regression that could have a huge fall-out: every single project using this or a similar snippet will need to manually add `--profile minimal`.

**Comment by kinnison** [comment]
Rustup 1.20.1 released this morning *should* solve that problem.  So long as Travis' rustup-init.sh fetches rustup-init from the proper release site, not their proxy, then it ought to be resolved by now.

**Comment by RalfJung** [comment]
I forgot to link to an affected build log: https://travis-ci.org/RalfJung/miri-test-libstd/builds/598465733

**Comment by kinnison** [comment]
Yeah that was a bug in 1.20.0 which is semi-resolved in 1.20.1 though #2065 was filed which suggests 1.20.1 didn't fully fix it, there's #2066 in test which hopefully will put this to bed and if so I'll prep 1.20.2 because I'm apparently a sucker for release numbers.

**Comment by RalfJung** [comment]
Great to hear that a fix is coming though. :)

**Comment by RalfJung** [comment]
But OTOH, looking at https://build.travis-ci.org/files/rustup-init.sh that seems to still be rustup 1.19 so this can't be a bug in 1.20.0...?

**Comment by kinnison** [comment]
The shell script is versioned separately from the installer binary, believe me 1.19 doesn't do profiles at all :D

**Comment by RalfJung** [comment]
Oh I do believe you. :) I hypothesized that something changed in the manifest that made old rustup versions install clippy by default as well now. But everything is more complicated as I thought, as always.^^

For now I am just using the upstream scripts, bypassing Travis' cache.

**Comment by kinnison** [comment]
I believe this is fixed after 1.20.2
